modules/openxr/editor/openxr_select_runtime.cpp

```diff
@@ -0,0 +1,124 @@
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "openxr_select_runtime.h"
+
+#include "core/io/dir_access.h"
+#include "core/os/os.h"
+#include "editor/editor_settings.h"
+#include "editor/editor_string_names.h"
+
+void OpenXRSelectRuntime::_bind_methods() {
+}
+
+void OpenXRSelectRuntime::_update_items() {
+	Ref<DirAccess> da = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+	OS *os = OS::get_singleton();
+	Dictionary runtimes = EDITOR_GET("xr/openxr/runtime_paths");
+
+	int current_runtime = 0;
+	int index = 0;
+	String current_path = os->get_environment("XR_RUNTIME_JSON");
+
+	// Parse the users home folder
+	String home_folder = os->get_environment("HOME");
+	if (home_folder == "") {
+		home_folder = os->get_environment("HOMEDRIVE") + os->get_environment("HOMEPATH");
+	}
+
+	clear();
+	add_item("Default", -1);
+	set_item_metadata(index, "");
+	index++;
+
+	Array keys = runtimes.keys();
+	for (int i = 0; i < keys.size(); i++) {
+		String key = keys[i];
+		String path = runtimes[key];
+		String adj_path = path.replace("~", home_folder);
+
+		if (da->file_exists(adj_path)) {
+			add_item(key, index);
+			set_item_metadata(index, adj_path);
+
+			if (current_path == adj_path) {
+				current_runtime = index;
+			}
+			index++;
+		}
+	}
+
+	select(current_runtime);
+}
+
+void OpenXRSelectRuntime::_item_selected(int p_which) {
+	OS *os = OS::get_singleton();
+
+	if (p_which == 0) {
+		// Return to default runtime
+		os->set_environment("XR_RUNTIME_JSON", "");
+	} else {
+		// Select the runtime we want
+		String runtime_path = get_item_metadata(p_which);
+		os->set_environment("XR_RUNTIME_JSON", runtime_path);
+	}
+}
+
+void OpenXRSelectRuntime::_notification(int p_notification) {
+	switch (p_notification) {
+		case NOTIFICATION_ENTER_TREE: {
+			// Update dropdown
+			_update_items();
+
+			// Connect signal
+			connect("item_selected", callable_mp(this, &OpenXRSelectRuntime::_item_selected));
+		} break;
+		case NOTIFICATION_EXIT_TREE: {
+			// Disconnect signal
+			disconnect("item_selected", callable_mp(this, &OpenXRSelectRuntime::_item_selected));
+		} break;
+	}
+}
+
+OpenXRSelectRuntime::OpenXRSelectRuntime() {
+	Dictionary default_runtimes;
+	default_runtimes["Monado (linux)"] = "/usr/share/openxr/1/openxr_monado.json";
+	default_runtimes["SteamVR (linux)"] = "~/.steam/steam/steamapps/common/SteamVR/steamxr_linux64.json";
+	default_runtimes["SteamVR (windows)"] = "C:\\Program Files (x86)\\Steam\\steamapps\\common\\SteamVR\\steamxr_win64.json";
+	default_runtimes["Meta (windows)"] = "C:\\Program Files\\Oculus\\Support\\oculus-runtime\\oculus_openxr_64.json";
+	default_runtimes["WMR"] = "C:\\WINDOWS\\system32\\MixedRealityRuntime.json";
```

Comment:
We should group them under `#ifdef` per platform, which would remove the need for the platform suffix in parentheses.

Reply:
Good idea, in the plugin we didn't have a choice but here that would work nicely.

It's a shame the Meta XR simulator isn't a separate install with a default installation path or we could add it in